libmscore/cmd.cpp

```diff
@@ -3893,6 +3893,17 @@ void Score::cmd(const QAction* a, EditData& ed)
             { "add-audio",                  [this]{ addAudioTrack();                                            }},
             { "transpose-up",               [this]{ transposeSemitone(1);                                       }},
             { "transpose-down",             [this]{ transposeSemitone(-1);                                      }},
+            {     // Diatonic up/down, keeping degree alterations:
+                  "pitch-up-diatonic-alterations", [this] {
+                        // Score::transpose() absolutely requires valid selection before invocation
+                        if (!score()->selection().isNone())
+                              score()->transpose(TransposeMode::DIATONICALLY, TransposeDirection::UP, Key::C, 1, true, true, true);
```

Comment:
Better put these into a separate function, even if they're only two lines. Otherwise code consistency is broken.
Same for the next one.

Reply:
Ah alright. No one wonder the commands in scoreview.cpp haven't been converted to commands in cmd.cpp haha (~127 in number.) Will do here in a moment.

Reply:
They're actually converted in https://github.com/musescore/MuseScore/compare/master...mattmcclinch:restructure-scoreview-cmd, but there's no PR yet.

Reply:
Hey thanks for the link. If that works out, it should be useful for further command implementations.

Reply:
There is now, see #5584